src/datasets/utils/file_utils.py

```diff
@@ -260,7 +259,10 @@ def get_authentication_headers_for_url(
         elif isinstance(token, str):
             token = token
         else:
-            token = HfFolder.get_token()
+            if config.HF_HUB_VERSION > version.parse("0.20.0"):
+                token = huggingface_hub.get_token()
+            else:
+                token = huggingface_hub.HfFolder.get_token()
```

Comment:
Actually the whole `get_authentication_headers_for_url` can be replaced by [`huggingface_hub.utils.build_hf_headers`](https://github.com/huggingface/huggingface_hub/blob/d72abd7ed4235a2b0d9edb422bc7edcd340687bc/src/huggingface_hub/utils/_headers.py#L40) which returns a dictionary with authorization + user_agent. You can then override some fields if needed.